docs/zdgbook/ObjectPublishing.rst

```diff
@@ -34,21 +34,31 @@ web publishing.
 HTTP Publishing
 ===============
 
-When you contact Zope with a web browser, your browser sends an HTTP
-request to Zope's web server. After the request is completely
-received, it is processed by 'ZPublisher', which is Zope's object
-publisher. 'ZPublisher' is a kind of light-weight ORB (Object Request
+Zope 4 no longer ships with a builtin web server, so when you want
+to interact with Zope via browser you have to setup a WSGI server as
+a middleware.
```

Comment:
I don't think that the server is a middleware. See [this graphical explanation](https://image.slidesharecdn.com/wsgi-150310012800-conversion-gate01/95/python-wsgi-introduction-3-1024.jpg?cb=1479297785 ) I found. (This image was taken from https://www.slideshare.net/AgeeleshwarK/python-wsgi)

Reply:
Just leave out the last part of that sentence, "as a middleware". My understanding is that "middleware" is software in the WSGI stack that the request passes through on the way to or from Zope.

Reply:
That's my understanding, too.

Reply:
@icemac and @dataflake - thank you for the time to review the PR and offer some improvements.

Obviously, I misunderstood the term "middleware" in this context - my thinking was that a middleware (alias for WSGI server e.g. gunicorn) is the part which manages the communication between the web server (eg. Nginx) and the application (eg. Zope/Flask).

After reading a couple of blog posts and the PEP 3333 I get a better picture.

Gonna update the text in a sec.

Thanks again for your time. I hope updating the documentation is a win-win for all of us - I learn a lot about the Zope stack and web development, and from the Zope side... the documentation gets updated for the first time in years :-)